scripts/create_burndown.py

```diff
@@ -17,211 +17,402 @@
+    os.chdir(args.output)
+  
+  if args.verbose:
+    print args
+
+  return args
+
+
+
+class BurndownData:
+  "Store burndown data parsed from YAML file"
+
+  # Constructor
+  def __init__(self, args):
+    burndown = self.openYAML(args.sprint)
+    self.getSprintData(burndown)
+
+  # Open burndown yaml file
+  def openYAML(self, sprint_number):
+    with open('burndown-data-' + sprint_number + '.yaml', 'r') as f:
+      burndown = yaml.load(f)
+    return burndown
+  
+  # Get data from burndown yaml file
+  def getSprintData(self, burndown):
+    current_day = 1
+  
+    self.sprint_number = burndown["meta"]["sprint"]
+    self.weekend_lines = burndown["meta"]["weekend_lines"]
+    self.total_days = burndown["meta"]["total_days"]
+    self.extra_day = 0
+    self.current_day = 1
+    self.days = []
+    self.extra_days = []
+    self.open_story_points = []
+    self.total_story_points = []
+    self.bonus_story_points_done = []
+    self.open_tasks = []
+    self.total_tasks = []
+    self.bonus_tasks_done = []
+    self.bonus_tasks_day_one = 0
+    self.x_fast_lane = []
+    self.y_fast_lane = []
+    self.total_fast_lane = []
+    self.max_story_points = 0
+  
+    for day in burndown["days"]:
+      self.days.append(current_day)
+      self.open_story_points.append(day["story_points"]["open"])
+      self.total_story_points.append(day["story_points"]["total"])
+      self.open_tasks.append(day["tasks"]["open"])
+      self.total_tasks.append(day["tasks"]["total"])
+      self.max_story_points = max(self.max_story_points, day["story_points"]["total"])
+    
+      # TODO: refactor to not to draw 0 extra Story Points Line if one task
+      # in extra card been done. Example http://imagebin.suse.de/1785/img
+      if "story_points_extra" in day or "tasks_extra" in day:
+        self.extra_days.append(current_day)
+        tasks = 0
+        if day.has_key("tasks_extra"):
+          tasks = -day["tasks_extra"]["done"]
+        self.bonus_tasks_done.append(tasks)
+        points = 0
+        if day.has_key("story_points_extra"):
+          points = -day["story_points_extra"]["done"]
+        self.bonus_story_points_done.append(points)
+    
+      if day.has_key("fast_lane"):
+        self.x_fast_lane.append(current_day)
+        self.y_fast_lane.append(day["fast_lane"]["open"])
+        self.total_fast_lane.append(day["fast_lane"]["total"])
+    
+      current_day += 1
+  
+    # If there are already done tasks at the beginning of a sprint
+    if burndown["days"][0].has_key("tasks_extra"):
+       self.bonus_tasks_day_one = burndown["days"][0]["tasks_extra"]["done"]
+    
+    # Add a day at the beginning of the extra days, so the curve starts at zero
+    if len(self.extra_days) > 0 and not self.bonus_tasks_day_one:
+      self.extra_days = [self.extra_days[0] - 1] + self.extra_days
+      self.bonus_story_points_done = [0] + self.bonus_story_points_done
+      self.bonus_tasks_done = [0] + self.bonus_tasks_done
+      self.extra_day = 1
```

Comment:
This is another example for calculation, which could be extracted to a separate function. This would make the code clearer, remove the need for a comment, and make it possible to test it separately in a unit test.